### DISCUSSION TITLE
Blog #3

### DISCUSSION THREAD

**Discussion by Unleqit** [AlinaBoess/SoftwareEngineeringProjektTINF23B5#3] [General]
# Dritter Blogeintrag

Schön, dass ihr es wieder zu uns geschafft habt :)
Hier sind ein paar Updates zu unserem Projekt:

## Verhaltensmodellierungen

Wir haben die Verhaltensmodellierungen unseres Projektes erstellt, welche nachfolgend eingebunden sind.

![Registrierung](https://github.com/AlinaBoess/SoftwareEngineeringProjektTINF23B5/blob/main/unfinishedBlogPosts/Register.png)

Bein Registrierungsvorgang wird ein Nutzer aufgefordert, dessen E-Mail Adresse sowie ein sicheres Passwort einzugeben. 
Im nächsten Schritt wird die Eingabe überprüft und an das Backend weitergeleitet; dieses gibt daraufhin entweder eine Bestätigung, oder eine Fehlermeldung zurück, welche anschließend im Frontend angezeigt wird.

![Anmelden](https://github.com/AlinaBoess/SoftwareEngineeringProjektTINF23B5/blob/main/unfinishedBlogPosts/Sign_IN.png)

Bein Anmeldevorgang wird ein Nutzer aufgefordert, dessen Anmeldedaten bestehend aus E-Mail Adresse sowie Passwort einzugeben. 
Im nächsten Schritt wird die Eingabe überprüft und an das Backend weitergeleitet; dieses gibt daraufhin entweder eine Bestätigung, oder eine Fehlermeldung zurück, welche anschließend im Frontend angezeigt wird.

![Reservierung](https://github.com/AlinaBoess/SoftwareEngineeringProjektTINF23B5/blob/main/unfinishedBlogPosts/Reservieren.png)

Bein Reservierungsvorgang wird ein Nutzer aufgefordert, dessen E-Mail Adresse sowie den Vor- und Nachnamen jener Person einzugeben, auf welche der Tisch reserviert wird.
Im nächsten Schritt wird die Eingabe überprüft und an das Backend weitergeleitet; dieses gibt daraufhin entweder eine Bestätigung, oder eine Fehlermeldung zurück, welche anschließend im Frontend angezeigt wird.

## Use-Case-Realization Specification (UCRS)

Des Weiteren haben wir ein [UCRS](https://github.com/AlinaBoess/SoftwareEngineeringProjektTINF23B5/blob/main/documentation/SRS/UCRS.md) erstellt, welche ihr gerne unter dem Hotlink einsehen könnt.

## Software Requirement Specification (SRS)

Unser im letzten Blog-Post vorgestelltes [SRS](https://github.com/AlinaBoess/SoftwareEngineeringProjektTINF23B5/blob/main/documentation/SRS/SoftwareRequirementDocumentation.md) haben wir um Aktivitätsdiagramme zu den jeweiligen Use Cases erweitert; dies könnt ihr euch ebenfalls gerne nochmals anschauen.

## Sonstige Aufgaben
Ansonsten haben wir uns noch hiermit beschäftigt:
- Weiterführende Strukturierung unseres GitHub Repositorys
- Detailliertere Organisation des Projektes in unserem Jira

Schaut gerne beim nächsten Mal wieder vorbei, wir freuen uns auf euch ;)
Bei Wünschen oder Anregungen dürft ihr uns gerne schreiben, wir freuen uns auf euer Feedback!

Liebe Grüße 
Lukas, Alex, Moumen, Yahya und Alina

**Comment by SchloesserJonas** [comment]
Guten Abend,
Das sieht schon mal solide aus!
Was ist der Hintergrund das ihr die Email abfragt? Gedenkt ihr eine Email als Bestätigung zu senden?
Was ist der Grund das ihr ein Single-Repo Konzept verfolgt (also Frontend, Backend und Docs in einem Repo habt)?

GaLiGrü 😊
Paula, Safae, Samuel, Valentin, Jonas

> Green Sprout 🌱

**Comment by Mimikry2** [comment]
Feedback zu den Use Cases "Reservierung vornehmen" und "Registrierung":

Moin,

eure Use-Case-Realization Specifications für „Reservierung vornehmen“ und „Registrierung“ sind klar strukturiert und bieten einen umfassenden Überblick über die jeweiligen Abläufe. Die detaillierte Beschreibung der Schritte im Flow of Events zeigt, wie durchdacht ihr die Benutzerinteraktionen geplant habt.

Besonders gut gefallen uns die Aspekte der Datensicherheit und Benutzerfreundlichkeit, die in beiden Dokumenten hervorgehoben werden. Die Echtzeitüberprüfung der Tischverfügbarkeit und die Validierung der Benutzereingaben sind entscheidende Faktoren für eine positive Benutzererfahrung.

Für den „Reservierung vornehmen“ Use Case wäre es hilfreich, mehr Informationen über die spezifischen Maßnahmen zur Sicherstellung der Datensicherheit und zum Umgang mit möglichen Fehlerszenarien zu erhalten. Im „Registrierung“ Use Case könnte eine genauere Beschreibung der Schritte zur Passwortsicherheit und -verschlüsselung dazu beitragen, das Vertrauen der Benutzer in das System zu stärken.

Zudem wäre es spannend zu erfahren, ob ihr an zusätzlichen Funktionen denkt, wie z.B. die Möglichkeit, Reservierungen zu bearbeiten oder Kontoinformationen zu aktualisieren, um die Benutzerfreundlichkeit weiter zu verbessern.

Wir freuen uns darauf zu sehen, wie sich euer Projekt weiterentwickelt!

Viele Grüße,
Tim, Dennis, Nico
